cmd/migrate.go

```diff
@@ -0,0 +1,287 @@
+		},
+	}
+
+	/*
+	 * CREATE TARGETS
+	 */
+
+	languages := generationWorkflow.Jobs.Generate.With["languages"].(string)
+	langToOutput, err := parseLanguages(languages)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	targets := map[string]workflow.Target{}
+	for lang, output := range langToOutput {
+		targets[fmt.Sprintf("%s-target", lang)] = workflow.Target{
+			Source:     defaultSourceName,
+			Target:     lang,
+			Output:     output,
+			Publishing: getPublishing(generationWorkflow, lang),
+		}
+	}
+
+	workflowFile := &workflow.Workflow{
+		Version: "1.0.0",
+		Sources: sources,
+		Targets: targets,
+	}
+
+	return workflowFile, newGenWorkflow, nil
+}
+
+func parseLanguages(langString string) (map[string]*string, error) {
+	languageElems, err := getAllSubmatches(regexp.MustCompile(`\s*?- (.*)`), langString)
+	if err != nil {
+		return nil, err
+	}
+
+	languageToSubdirectory := map[string]*string{}
+	for _, languageElem := range languageElems {
+		// Languages are in one of the following formats:
+		// - go: ./path-to-go
+		// - go
+		if strings.Contains(languageElem, ":") {
+			languageAndSubdirectory := strings.Split(languageElem, ": ")
+			if len(languageAndSubdirectory) != 2 { //nolint:gomnd
+				return nil, fmt.Errorf("incorrectly formatted language in workflow file. Culprit: %s", languageElem)
+			}
+			languageToSubdirectory[languageAndSubdirectory[0]] = &languageAndSubdirectory[1]
+		} else {
+			languageToSubdirectory[languageElem] = nil
+		}
+	}
+
+	return languageToSubdirectory, nil
+}
+
+func docLocationsToDocuments(docLocations []string, authHeader string, hasToken bool) []workflow.Document {
+	var auth *workflow.Auth
+	if hasToken {
+		auth = &workflow.Auth{
+			Header: authHeader,
+			Secret: "$OPENAPI_DOC_AUTH_TOKEN", // This is the name of the secret in the action, which is fixed
+		}
+	}
+
+	var documents []workflow.Document
+	for _, docLocation := range docLocations {
+		documents = append(documents, workflow.Document{
+			Location: docLocation,
+			Auth:     auth,
+		})
+	}
+	return documents
+}
+
+func getPublishing(genWorkflow config.GenerateWorkflow, lang string) *workflow.Publishing {
+	pubVal, ok := genWorkflow.Jobs.Generate.With[fmt.Sprintf("publish_%s", lang)]
+	shouldPublish := ok && pubVal.(bool)
+
+	if shouldPublish {
+		// These secret values are hardcoded because they are the names of the secrets in the action
+		switch lang {
+		case "typescript":
+			return &workflow.Publishing{
+				NPM: &workflow.NPM{
+					Token: "$NPM_TOKEN",
+				},
+			}
+		case "python":
+			return &workflow.Publishing{
+				PyPi: &workflow.PyPi{
+					Token: "$PYPI_TOKEN",
+				},
+			}
+		case "php":
+			return &workflow.Publishing{
+				Packagist: &workflow.Packagist{
+					Username: "$PACKAGIST_USERNAME",
+					Token:    "$PACKAGIST_TOKEN",
+				},
+			}
+		case "java":
+			return &workflow.Publishing{
+				Java: &workflow.Java{
+					OSSRHUsername: "$OSSRH_USERNAME",
```

Comment:
In the new workflow the user can technically set these environment variables to whatever they want. Are we assuming that in existing generation actions they will always be consistent.